### PR TITLE
Subparts can track separate lists of sublocations that exist bellow them

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
None

#### Purpose of change
No one has made an issue but multiple people have reported a bug where binoculars are always uncomfortable to wear. This is because there is no relation between the secondary hanging locations and primary locations like upper and lower torso. 

Supersedes #61545 

#### Describe the solution
Subparts can now specify an array of other subparts that exist bellow them. This is the list of location tested against when checking for comfort / discomfort.

Added a message letting you know when you wear something what is chaffing, where and why.

#### Testing
make the stethoscope weigh more than it does and try putting it on with a belly wrap 
image

binoculars alone
image

binoculars after wearing a sweater
image

binoculars with only a belly wrap
image

#### Additional context
The location choices are:
Neck -> upper torso
Hanging Front -> upper and lower torso
Hanging Back -> upper and lower torso
Belt -> the hips
Draped legs -> the hips and upper legs

### CODE REVIEW

File: src/character_attire.cpp [hunk 1]
```diff
@@ -1889,8 +1889,17 @@ std::unordered_set<bodypart_id> outfit::where_discomfort() const
             // note anything selectively rigid reasonably can be assumed to support itself so we don't need to worry about this
             // items must also be somewhat heavy in order to cause discomfort
             if( !i.is_bp_rigid_selective( sbp ) && !i.is_bp_comfortable( sbp ) &&
-                covered_sbps.count( sbp ) != 1 && i.weight() > units::from_gram( 250 ) ) {
-                uncomfortable_bps.insert( sbp->parent );
+                i.weight() > units::from_gram( 250 ) ) {
+
+                // need to go through each locations under location to check if its covered, since secondary locations can cover multiple underlying locations
+                for( const sub_bodypart_id &under_sbp : sbp->locations_under ) {
```
Comment (dseguin) [inline src/character_attire.cpp:1895]:
```suggestion
                for( const sub_bodypart_str_id &under_sbp : sbp->locations_under ) {
```
Small fix for Clang 12

Comment (bombasticSlacks) [reply src/character_attire.cpp]: thanks

File: src/character_attire.cpp [hunk 2]
```diff
@@ -1889,8 +1889,17 @@ std::unordered_set<bodypart_id> outfit::where_discomfort() const
...
             if( !i.is_bp_rigid_selective( sbp ) && !i.is_bp_comfortable( sbp ) &&
-                covered_sbps.count( sbp ) != 1 && i.weight() > units::from_gram( 250 ) ) {
-                uncomfortable_bps.insert( sbp->parent );
+                i.weight() > units::from_gram( 250 ) ) {
+
+                // need to go through each locations under location to check if its covered, since secondary locations can cover multiple underlying locations
+                for( const sub_bodypart_str_id &under_sbp : sbp->locations_under ) {
+                    if( covered_sbps.count( under_sbp ) != 1 ) {
+                        guy.add_msg_if_player(
+                            string_format( "<color_c_red> the %s rubs uncomfortably against your unpadded %s </color>",
```
Comment (dseguin) [inline src/character_attire.cpp:1898]:
```suggestion
                            string_format( _( "<color_c_red> the %s rubs uncomfortably against your unpadded %s </color>" ),
```

Comment (bombasticSlacks) [reply src/character_attire.cpp]: wow sorry I missed this issue my bad for posting in ready to merge

**Pull request merged**